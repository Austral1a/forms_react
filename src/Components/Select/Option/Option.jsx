import React from 'react'
import classNames from 'classnames/dedupe'

export function Option({
    value,
    text,
    className
}) {
    const customClasses = classNames('select__option', className)
    return (
        <option
            className={customClasses}
            value={value}
        >
            {text}
        </option>
    )
}

