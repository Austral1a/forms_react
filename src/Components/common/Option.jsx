import React from 'react'

function Option({
    value,
    selected,
    text
}) {
    return (
        <option
            value={value}
            selected={selected}
        >
            {text}
        </option>
    )
}

export default Option
