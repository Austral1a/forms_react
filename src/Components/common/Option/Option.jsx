import React from 'react'
import classNames from 'classnames/dedupe'

function Option({
    value,
    text,
    className
}) {
    const customClasses = classNames(className)
    return (
        <option
            className={customClasses}
            value={value}
        >
            {text}
        </option>
    )
}

export default Option
