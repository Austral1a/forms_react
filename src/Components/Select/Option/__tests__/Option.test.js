import {shallow} from 'enzyme';
import {Option} from '../Option';
import toJson from 'enzyme-to-json';

describe('Option component', () => {
    it('Option must match snapshot', () => {
        const wrapper = shallow(<Option />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("must render option's inner text", () => {
        const text = 'this is options';
        const wrapper = shallow(<Option text={text} />);
        expect(wrapper.find('.select__option').text()).toBe(text);
    });

    it("option must have value attribute", () => {
        const value = 'option1';
        const wrapper = shallow(<Option value={value} />);
        expect(wrapper.find('.select__option').prop('value')).toBe(value);
    });
});
