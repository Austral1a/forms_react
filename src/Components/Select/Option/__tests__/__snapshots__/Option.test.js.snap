// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option component Option must match snapshot 1`] = `
<option
  className="select__option"
/>
`;
